src/hivpy/hiv_testing.py

```diff
@@ -0,0 +1,145 @@
+
+        # eff_max_freq_testing is used as an index to pick the correct
+        # minimum number of days to wait between tests from this list
+        self.days_to_wait = [365, 180, 90]  # 12 months, 6 months, 3 months
+
+        self.rate_first_test = 0
+        self.rate_rep_test = 0
+        # TODO: meant to be a personal variable but currently always just set to test_targeting
+        self.eff_test_targeting = self.test_targeting
+        # scenario has a high chance to change eff_test_targeting
+        if self.test_scenario == 1:
+            r = rng.uniform()
+            if r < 0.45:
+                self.eff_test_targeting = 2
+            elif r < 0.9:
+                self.eff_test_targeting = 5
+        # FIXME: move this to a yaml later
+        self.covid_disrup_affected = False
+        self.testing_disrup_covid = False
+
+    def update_hiv_testing(self, pop):
+        """
+        Update which individuals in the population have been tested.
+        COVID disruption is factored in.
+        """
+        # testing occurs after a certain year if there is no covid disruption
+        if ((pop.date.year >= (self.date_start_testing + 5.5))
+           & ((not self.covid_disrup_affected) | (not self.testing_disrup_covid))):
+
+            # update testing probabilities
+            self.rate_first_test = self.init_rate_first_test + (min(pop.date.year, self.date_test_rate_plateau)
+                                                                - (self.date_start_testing + 5.5)) \
+                                                                * self.an_lin_incr_test
+            self.rate_rep_test = (min(pop.date.year, self.date_test_rate_plateau)
+                                  - (self.date_start_testing + 5.5)) * self.an_lin_incr_test
+
+            # get population ready for testing
+            testing_population = pop.get_sub_pop([(col.HARD_REACH, op.eq, False),
+                                                  (col.AGE, op.ge, 15),
+                                                  (col.DATE_OF_DEATH, op.eq, None),
+                                                  (col.HIV_STATUS, op.eq, False),
+                                                  [(col.LAST_TEST_DATE, op.le, pop.date -
+                                                    timedelta(days=self.days_to_wait[self.eff_max_freq_testing])),
+                                                   (col.LAST_TEST_DATE, op.eq, None)]
+                                                  ])
+
+            # first time testers
+            untested_population = pop.apply_bool_mask(~pop.data.loc[testing_population, col.EVER_TESTED],
+                                                      testing_population)
+            # repeat testers
+            prev_tested_population = pop.apply_bool_mask(pop.data.loc[testing_population, col.EVER_TESTED],
+                                                         testing_population)
```

Comment:
```suggestion
# first time testers
            untested_population = pop.apply_bool_mask(~pop.get_variable(col.EVER_TESTED, testing_population),
                                                      testing_population)
            # repeat testers
            prev_tested_population = pop.apply_bool_mask(pop.get_variable(col.EVER_TESTED, testing_population),
                                                         testing_population)
```
To avoid direct `.data` accesses